src/shared/fdset.c

```diff
@@ -97,21 +97,33 @@ int fdset_consume(FDSet *s, int fd) {
         return r;
 }
 
-int fdset_put_dup(FDSet *s, int fd) {
+int fdset_put_dup_full(FDSet *s, int fd, bool store_index) {
```

Comment:
is `store_index` the right name for this? i mean, you are not storing an index here, you